scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java

```diff
@@ -806,42 +812,39 @@ private void initSelectWorkflowFromCatalogPanel() {
-                            for (int i = 0; i < workflowsSize; i++) {
-                                JSONObject workflow = workflows.get(i).isObject();
-                                String workflowName = workflow.get(KEY_OF_NAME).isString().stringValue();
-                                workflowsListBox.addItem(workflowName);
-                            }
-                            workflowsListBox.setEnabled(true);
-                        }
-
-                        @Override
-                        public void onError(Request request, Throwable exception) {
-                            GWT.log("OOPS:" + request.toString());
+        bucketsListBox.addChangeHandler(event -> {
+            String selectedBucket = bucketsListBox.getSelectedValue();
+            if (!CATALOG_SELECT_BUCKET.equals(selectedBucket)) {
+                String workflowUrl = CATALOG_URL + "/buckets/" + selectedBucket + "/resources?kind=workflow";
+                RequestBuilder req = new RequestBuilder(RequestBuilder.GET, workflowUrl);
+                req.setHeader(SESSION_ID_PARAMETER_NAME, LoginModel.getInstance().getSessionId());
+                req.setCallback(new RequestCallback() {
+                    @Override
+                    public void onResponseReceived(Request request, Response response) {
+                        JSONArray workflows = JSONParser.parseStrict(response.getText()).isArray();
+                        int workflowsSize = workflows.size();
+                        workflowsListBox.setEnabled(false);
+                        workflowsListBox.clear();
+                        workflowsListBox.addItem(CATALOG_SELECT_WF);
+                        for (int i = 0; i < workflowsSize; i++) {
+                            JSONObject workflow = workflows.get(i).isObject();
+                            String workflowName = workflow.get(KEY_OF_NAME).isString().stringValue();
+                            workflowsListBox.addItem(workflowName);
                         }
-                    });
-                    try {
-                        req.send();
-                    } catch (RequestException e) {
-                        GWT.log("Error occured when fetching workflows from Catalog");
-                        e.printStackTrace();
+                        workflowsListBox.setEnabled(true);
+                    }
 
+                    @Override
+                    public void onError(Request request, Throwable exception) {
+                        LogModel.getInstance().logImportantMessage("OOPS:" + request.toString());
                     }
+                });
+                try {
+                    req.send();
+                } catch (RequestException e) {
+                    LogModel.getInstance().logImportantMessage("Error occured when fetching workflows from Catalog");
+                    // For Dev debug mode only
+                    e.printStackTrace();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1148)